scripts/mavi-backup.sh

```
@@ -0,0 +1,115 @@
+
+if [[ -e "$backup_dir" ]]; then
+  if [[ ! -d "$backup_dir" ]]; then
+    printf 'Backup target exists and is not a directory: %s\n' "$backup_dir" >&2
+    exit 1
+  fi
+  if [[ -n "$(find "$backup_dir" -mindepth 1 -maxdepth 1 -print -quit)" ]]; then
+    printf 'Backup target is not empty: %s\n' "$backup_dir" >&2
+    exit 1
+  fi
+else
+  mkdir -p -m 700 "$backup_dir"
```

Shellcheck (0.11.0)

[warning] When used with -p, -m only applies to the deepest directory.

(SC2174)